e2e/test/helpers/TestDevice.cs

```diff
@@ -167,24 +167,6 @@ public string ConnectionString
 
         public Client.IAuthenticationMethod AuthenticationMethod { get; private set; }
 
-        public IotHubDeviceClient CreateDeviceClient(IotHubClientOptions options = default)
```

Comment:
CreateDeviceClient had the following overloads:
public IotHubDeviceClient CreateDeviceClient(IotHubClientOptions options = default)
public IotHubDeviceClient CreateDeviceClient(IotHubClientOptions options = default, ConnectionStringAuthScope authScope = ConnectionStringAuthScope.Device)

They compile to the same code, causing ambiguity when CreateDeviceClient() is called. So, I'm removing the redundant overload.